packages/fmg-core/test/framework/transitions.js

```diff
@@ -0,0 +1,323 @@
+import assertRevert from '../helpers/assert-revert';
+
+import { CountingGame } from '../test-game/src/counting-game';
+import { Channel, State } from '../..';
+
+var CommonState = artifacts.require("./CommonState.sol");
+var Framework = artifacts.require("./Framework.sol");
+var CountingStateContract = artifacts.require("../test-game/contracts/CountingState.sol");
+var CountingGameContract = artifacts.require("../test-game/contracts/CountingGame.sol");
+
+contract('Framework', (accounts) => {
+    let channel, otherChannel, defaults, framework;
+    const resolution = [12, 13];
+    const otherResolution = [10, 15];
+
+    let fromState, toState;
+
+    before(async () => {
+        framework = await Framework.deployed();
+
+        CountingStateContract.link(CommonState);
+        let stateContract = await CountingStateContract.new();
+        CountingGameContract.link("CountingState", stateContract.address);
+        let gameContract = await CountingGameContract.new();
+
+        channel = new Channel(gameContract.address, 0, [accounts[0], accounts[1]]);
+        otherChannel = new Channel(gameContract.address, 1, [accounts[0], accounts[1]]);
+
+        let challengeeBal = Number(web3.toWei(6, "ether"));
+        let challengerBal = Number(web3.toWei(4, "ether"));
+        
+        defaults = { channel, resolution, gameCounter: 0};
+    });
+
+
+    const validTransition = async (state1, state2) => { 
+        return await framework.validTransition(state1.toHex(), state2.toHex());
+    };
+
+    describe("propose -> propose", () => {
+        beforeEach(() => {
+            fromState= CountingGame.proposeState({ ...defaults, turnNum: 0, stateCount: 0 });
+            toState = CountingGame.proposeState({ ...defaults, turnNum: 1, stateCount: 1 });
+        });
+
+        it("allows a valid transition", async () => {
+            assert(await validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the turnNum doesn't increment", async () => {
+            toState.turnNum = fromState.turnNum;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects any transition where the channel changes", async () => {
+            toState.channel = otherChannel;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the balances changes", async () => {
```

Comment:
I thought that `resolution` might potentially be closer to what we're talking about here - it's how much you get if the game ends at this point. To me `balances` is making some comment about how much you currently have, which seems like it might be worth avoiding as a concept here.